crates/forge/src/lib.rs

```diff
@@ -188,6 +195,81 @@ fn warn_if_available_gas_used_with_incompatible_scarb_version(
     Ok(())
 }
 
+#[derive(Default)]
+struct RpcDescriptor<'a> {
+    scarb_names: Vec<&'a str>,
+    test_paths: Vec<&'a str>,
+}
+
+async fn warn_if_incompatible_rpc_version(
+    test_crates: &[CompiledTestCrateRaw],
+    fork_targets: &[ForkTarget],
+) -> Result<()> {
+    let mut descriptors = HashMap::<&str, RpcDescriptor>::new();
+
+    for (raw_fork_config, test_case_name) in test_crates.iter().flat_map(|ctc| {
+        ctc.test_cases
+            .iter()
+            .filter(|tc| !tc.ignored)
+            .filter_map(|tc| tc.fork_config.as_ref().map(|fc| (fc, tc.name.as_str())))
+    }) {
+        let params = replace_id_with_params(raw_fork_config, fork_targets)?;
+        let descriptor = descriptors.entry(&params.url).or_default();
+
+        match raw_fork_config {
+            RawForkConfig::Id(name) => {
+                descriptor.scarb_names.push(name);
+            }
+            RawForkConfig::Params(_) => {
+                descriptor.test_paths.push(test_case_name);
+            }
+        };
+    }
+
+    for (url, descriptor) in descriptors {
+        const EXPECTED: &str = "0.6.0";
```

Comment:
I would also add an instruction how to update it somewhere in our development docs.
Also, a check before releasing in our release script, 
"Did you update `starknet.rs` package during this release cycle? (y/n) -> What is the spec version it supports? (input...)" 

Can be added as a separate issue since it's a bit extra.

Reply:
Created issue for this https://github.com/foundry-rs/starknet-foundry/issues/1660